src/lib/models/invoice.js

```diff
@@ -63,17 +75,82 @@ class Invoice {
     return this._address;
   }
 
-  set invoiceLicences (arr) {
-    assert(isArray(arr), 'Array expected');
-    arr.map((invoiceLicence, i) =>
-      assert(invoiceLicence instanceof InvoiceLicence, `InvoiceLicence expected at position ${i}`)
-    );
-    this._invoiceLicences = arr;
+  set invoiceLicences (invoiceLicences) {
+    assertIsArrayOfType(invoiceLicences, InvoiceLicence);
+    this._invoiceLicences = invoiceLicences;
   }
 
   get invoiceLicences () {
     return this._invoiceLicences;
   }
+
+  /**
+   * Gets an object containing the total values and number
+   * of charge types for this invoice in the shape:
+   *
+   * {
+   *   totalValue: Number,
+   *   totalInvoices: Number,
+   *   totalCredits: Number,
+   *   numberOfInvoices: Number,
+   *   numberOfCredits: Number
+   * }
+   */
+  getTotals () {
+    const transactions = this.invoiceLicences.reduce((acc, invoiceLicence) => {
+      return [...acc, ...invoiceLicence.transactions];
+    }, []);
+
+    return transactions.reduce((totals, { isCredit, value }) => {
+      const change = getTotalsChange(totals, value, isCredit);
+      return { ...totals, ...change };
+    }, {
+      totalValue: 0,
+      totalInvoices: 0,
+      totalCredits: 0,
+      numberOfInvoices: 0,
+      numberOfCredits: 0
+    });
+  }
+
+  /**
+   * Get the licence numbers for this invoice by inspecting
+   * the Licence objects associated with each InvoiceLicence
+   * object associated with this invoice
+   */
+  getLicenceNumbers () {
+    return flatMap(this.invoiceLicences, invoiceLicence => {
+      return get(invoiceLicence, 'licence.licenceNumber');
+    });
+  }
+
+  /**
+   * Gets the number of pence that are required to add to the invoice
+   * to satisfy the minimum charge of £25.
+   */
+  getMinimumPaymentTopUp () {
+    const { totalValue } = this.getTotals();
+    return totalValue < MINIMUM_PAYMENT ? MINIMUM_PAYMENT - totalValue : 0;
+  }
+
+  getInvoiceLicenceContacts () {
```

Comment:
What are these contacts being used for?  If it's for this view:
https://wrls-prototype.herokuapp.com/bd/charges/supplementary-charges
I think we should be using the invoice account holder company.
Also the invoice licence `contact` is not populated where the CRM `company` is a person - not sure if this has an impact on the code below.